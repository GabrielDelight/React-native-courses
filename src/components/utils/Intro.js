import React from 'react';
import {View, Text} from 'react-native';
const Intro = () => {
  return (
    <View>
      <Text>Hellow world!!</Text>
    </View>
  );
};

export default Intro;
