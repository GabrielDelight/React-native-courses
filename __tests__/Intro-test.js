import React from 'react';
import {Text, FlatList} from 'react-native';
import renderer from 'react-test-renderer';
import Intro from '../src/components/utils/Intro';

test('render correctly', () => {
  const tree = renderer.create(<Intro />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Render Flatlist Correctly', () => {
  const tree = renderer.create(
    <FlatList
      data={['John', 'Mike', 'Jack']}
      keyExtractor={(item) => item}
      renderItem={({item}) => <Text>{item}</Text>}
    />,
  ).toJSON()
  expect(tree).toMatchSnapshot()
});
