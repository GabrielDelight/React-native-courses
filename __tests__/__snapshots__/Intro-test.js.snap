// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Flatlist Correctly 1`] = `
<RCTScrollView
  data={
    [
      "John",
      "Mike",
      "Jack",
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        John
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Mike
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Jack
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`render correctly 1`] = `
<View>
  <Text>
    Hellow world!!
  </Text>
</View>
`;
